src/main/java/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,8 +16,9 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDb = userService.findByLogin(login);
-
-        if (userFromDb.isPresent() && userFromDb.get().getPassword().equals(password)) {
+        byte[] salt = userFromDb.get().getSalt();
+        String saltedPass = HashUtil.hashPassword(password, salt);
+        if (saltedPass.equals(userFromDb.get().getPassword())) {
```

Comment:
why are you no more checking if the value is present?

Reply:
Idea suggested me to remove it as it always either true or false